### PR TITLE
0.8.2 (Funnel Cake)

### PR DESCRIPTION
**Pull request by jasoncalabrese**
10624801_10203819110930588_2956395634112010527_n

* New Basal and Temp basal visualization with Care Portal and OpenAPS support
* Care Portal event type selection adjusts what fields are displayed
* Event types with duration (Temp Basal/Exercise/Notes)
* New scaling options: Logarithmic (how it's always been), Linear, and Logarithmic (Dynamic)
* New Admin tools to reduce the need to go into mongolab for common support issues
* Fix for iOS9 angle arrows
* BWP tooltip updates
* Chart time localizations
* Fix for Treatment report date filter
* Translation updates for Romanian, Norwegian, Bulgarian, Swedish, and Czech

Full list of merged pull requests: https://github.com/nightscout/cgm-remote-monitor/pulls?q=milestone%3A0.8.2+is%3Aclosed+is%3Apr

### CODE REVIEW

**Comment by MilosKozak** [comment]
:+1:

File: lib/client/renderer.js [hunk 1]
```diff
@@ -37,6 +39,12 @@ function init (client, d3) {
     return radius / focusRangeAdjustment();
   };
 
+  function tooltipLeft ( ) {
+    var windowWidth = $(client.tooltip).parent().parent().width();
+    var left = d3.event.pageX + TOOLTIP_WIDTH < windowWidth ? d3.event.pageX : windowWidth - TOOLTIP_WIDTH - 10;
+    return left + 'px'
```
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1231252424/issues/source?bid=2452374&fileBranchId=2932983#l45)

File: static/report/js/report.js [hunk 2]
```diff
@@ -388,24 +363,23 @@
     
     function daysfilter() {
       matchesneeded++;
-      for (var d in daystoshow) {
-        if (daystoshow.hasOwnProperty(d)) {
-          var day = new Date(d).getDay();
-          if (day===0 && $('#rp_su').is(':checked')) { daystoshow[d]++; }
-          if (day===1 && $('#rp_mo').is(':checked')) { daystoshow[d]++; }
-          if (day===2 && $('#rp_tu').is(':checked')) { daystoshow[d]++; }
```
Comment (jasoncalabrese) [inline static/report/js/report.js]: Codacy Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1231252485/issues/source?bid=2452374&fileBranchId=2932983#l366)

File: lib/client/ticks.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      , opts.targetBottom
+      , 120
+      , opts.targetTop
+      , opts.high
+      , 400
+    ];
+  }
+}
+
+function prepareLinear (client, opts) {
```
Comment (jasoncalabrese) [inline lib/client/ticks.js]: Codacy Issue found: ['opts' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1231252425/issues/source?bid=2452374&fileBranchId=2932983#l50)

**Pull request merged**